### PR TITLE
Update with Mountpoint 1.6.0 features

### PR DESCRIPTION
**Pull request by jjkr**
*Issue #, if available:* #153

*Description of changes:* Add examples and support for sse and max attempts options. Support for sse does not require any code changes because it is just an argument passthrough. The max attempts option is passed as an environment variable to mountpoint, so some conversion is needed from the mount option.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/driver/mount.go [hunk 1]
```diff
@@ -33,16 +33,18 @@ import (
...
+	accessKeyEnv         = "AWS_SECRET_ACCESS_KEY"
+	regionEnv            = "AWS_REGION"
+	defaultRegionEnv     = "AWS_DEFAULT_REGION"
+	stsEndpointsEnv      = "AWS_STS_REGIONAL_ENDPOINTS"
+	MountS3PathEnv       = "MOUNT_S3_PATH"
+	awsMaxAttemptsEnv    = "AWS_MAX_ATTEMPTS"
+	defaultMountS3Path   = "/usr/bin/mount-s3"
+	procMounts           = "/host/proc/mounts"
+	userAgentPrefix      = "--user-agent-prefix"
+	awsMaxAttemptsOption = "--aws-max-attempts"
```
Comment (dannycjones) [inline pkg/driver/mount.go:46]: Why are we adding the max attempts configuration as an option as well as an environment variable? What makes it different from something like `AWS_REGION` or `AWS_ACCESS_KEY_ID`?

Comment (jjkr) [reply pkg/driver/mount.go:46]: We need max attempts to be configurable by the user on a per PersistentVolume basis, so we have to accept it similar to other mount options. Unlike the other mount options, it's not just passed through to mountpoint and we have to do a conversion to remove the option from the CLI flags and pass it as an environment variable instead.

`AWS_REGION` we get from the driver's environment and that's passed through unchanged to mountpoint. `AWS_ACCESS_KEY_ID` is also unique where we get the key from a k8s secret and pass that through to the driver. Overall there's just different ways that the CSI driver gets options with different ways mountpoint accepts them and the driver has to glue everything together.

**Review by dannycjones — CHANGES_REQUESTED** [review]
Docs change, then lgtm

**Review by dannycjones — APPROVED** [review]
LGTM!

**Pull request merged**